modules/apps/transfer/types/transfer_authorization_test.go

```diff
@@ -190,6 +230,13 @@ func (suite *TypesTestSuite) TestTransferAuthorizationValidateBasic() {
 			},
 			true,
 		},
+		{
+			"success: with max Int allocation spend limit",
+			func() {
+				transferAuthz.Allocations[0].SpendLimit = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(math.MaxInt64)))
+			},
+			true,
+		},
```

Comment:
can you add a failure test case where the amount is greater than `math.Int64`? You can set it using 

```golang
sdk.NewIntFromBigInt(new(big.Int).SetUint64(math.MaxUint64))
```

Reply:
should this fail or should it behave as a normal spend limit and have the transfer token amount deducted? 

At the moment, the code panic's due to calling `Int64()` on the amount [here](https://github.com/cosmos/ibc-go/pull/3454/files#diff-44955ffabce2c0f3471ea97e335ca2a5e05b70a4349dca81e707b097292b7b07R51) when feeding in `sdk.NewIntFromBigInt(new(big.Int).SetUint64(math.MaxUint64))` in a testcase. This is expected according to `sdk.Int` godoc for `Int64()`: `panics if out of range`

Adapting the code to use the below can avoid panics, but then treats values greater than MaxInt64 as valid cases, which imo doesn't make sense as you can end up deducting a transfer amount amount from greater than MaxInt64 and landing on the unlimited spend limit sentinel value accidentally

```diff
+ if coin.Amount.Equal(sdkmath.NewInt(math.MaxInt64)) {
- if coin.Amount.Int64() == math.MaxInt64 {
	continue
}
```

Reply:
@damiannolan it seems like the `SpendLimit` can accept a value of size `uint256` using `sdk.NewIntFromBigInt(MaxUint256)`. Do you see this as problematic? On our end this works well since Solidity developers can directly pass `max(uint256)`. I've updated the PR to reflect this change.